src/Core/tests/UnitTests/ImageSource/ImageSourceServiceTests.cs

```diff
@@ -12,6 +12,61 @@ namespace Microsoft.Maui.UnitTests.ImageSource
 	[Category(TestCategory.Core, TestCategory.ImageSource)]
 	public class ImageSourceServiceTests
 	{
+		[Theory]
+		[InlineData(true, false)]
+		[InlineData(false, true)]
+		[InlineData(true, true)]
+		public void ResolvesImageSourceServiceInTypicalScenarios(bool registerInterfaces, bool registerClasses)
+		{
+			var provider = CreateImageSourceServiceProvider(services =>
+			{
+				if (registerInterfaces)
+				{
+					services.AddService<IStreamImageSource, StreamImageSourceService>();
+					services.AddService<IFileImageSource, FileImageSourceService>();
+				}
```

Comment:
Wouldn't this be redundant? `MauiApp.CreateBuilder()` will call https://github.com/dotnet/maui/blob/cb1ac24b9a52394169e5d2a0787e7f08543ff9b9/src/Core/src/Hosting/MauiAppBuilder.cs#L37, so the interfaces are always registered.

Reply:
You're right, I didn't realize that we always use the defaults. I'll replace the interfaces and classes with custom ones that aren't registered by default.